recipes/libccd/all/conanfile.py

```diff
@@ -25,62 +27,55 @@ class LibccdConan(ConanFile):
         "enable_double_precision": False,
     }
 
-    exports_sources = "CMakeLists.txt"
-    generators = "cmake"
-    _cmake = None
-
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
-
-    @property
-    def _build_subfolder(self):
-        return "build_subfolder"
-
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
     def configure(self):
         if self.options.shared:
             del self.options.fPIC
-        del self.settings.compiler.libcxx
-        del self.settings.compiler.cppstd
+        try:
+           del self.settings.compiler.libcxx
+        except Exception:
+           pass
+        try:
+           del self.settings.compiler.cppstd
+        except Exception:
+           pass
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-                  destination=self._source_subfolder, strip_root=True)
+        get(self, **self.conan_data["sources"][self.version],
+            destination=self.source_folder, strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["BUILD_DOCUMENTATION"] = False
+        tc.variables["ENABLE_DOUBLE_PRECISION"] = self.options.enable_double_precision
+        tc.variables["CCD_HIDE_ALL_SYMBOLS"] = not self.options.shared
+        tc.cache_variables["CMAKE_POLICY_DEFAULT_CMP0077"] = "NEW"
```

Comment:
So we dont need to git blame and read the PR history next time

```suggestion
        tc.cache_variables["CMAKE_POLICY_DEFAULT_CMP0077"] = "NEW" # https://github.com/conan-io/conan/issues/11840
```

Reply:
done